sw/device/lib/base/hardened_memory.c

```diff
@@ -106,7 +103,7 @@ void hardened_memshred(uint32_t *dest, size_t word_len) {
         ct_cmovw(ct_sltuw(launderw(byte_idx), byte_len), datap, decoy));
 
     // Write a freshly-generated random word to `*data`.
-    write_32(hardened_memshred_random_word(), data);
+    write_32(ibex_rnd_uint32(), data);
```

Comment:
Would hang at runtime if EDN isn't up; how do we know it is? If the caller needs to check EDN first, the documentation of the function should say so.